tsdb/chunkenc/histogram.go

```diff
@@ -1203,6 +1204,20 @@ func (it *histogramIterator) Next() ValueType {
 		} else {
 			it.nBuckets = nil
 		}
+		if len(it.pSpans) > 0 {
```

Comment:
I think this should be outside the `if it.atHistogramCalled`. PromQL engine typically calls this iterator with `AtFloatHistogram`. Please add a test case for calling that, basically copy paste the one you have now `TestHistogramUniqueSpansAfterNext`, but call `AtFloatHistogram`.